src/Harmony/StakingTransaction.h

```diff
@@ -0,0 +1,216 @@
+// Copyright © 2017-2019 Trust.
+//
+// This file is part of Trust. The full Trust copyright notice, including
+// terms governing use, modification, and redistribution, is contained in the
+// file LICENSE at the root of the source code distribution tree.
+
+#pragma once
+
+#include <boost/algorithm/string.hpp>
+#include <cstdint>
+#include <math.h>
+#include <stdint.h>
+#include <stdio.h>
+#include <string>
+#include <vector>
+
+#include "Address.h"
+#include "../uint256.h"
+using namespace std;
+
+namespace TW::Harmony {
+
+template <typename Directive>
+class StakingTransaction {
+  public:
+    uint8_t directive;
+    Directive stakeMsg;
+    uint256_t nonce;
+    uint256_t gasPrice;
+    uint256_t gasLimit;
+
+    // Signature values
+    uint256_t v = uint256_t();
+    uint256_t r = uint256_t();
+    uint256_t s = uint256_t();
+
+    StakingTransaction(uint8_t directive, Directive stakeMsg, uint256_t nonce, uint256_t gasPrice,
+                       uint256_t gasLimit, uint256_t v, uint256_t r, uint256_t s)
+        : directive(move(directive))
+        , stakeMsg(move(stakeMsg))
+        , nonce(move(nonce))
+        , gasPrice(move(gasPrice))
+        , gasLimit(move(gasLimit))
+        , v(move(v))
+        , r(move(r))
+        , s(move(s)) {}
+};
+
+enum Directive : uint8_t {
+    DirectiveCreateValidator,
+    DirectiveEditValidator,
+    DirectiveDelegate,
+    DirectiveUndelegate,
+    DirectiveCollectRewards
+};
+
+class Description {
+  public:
+    string name;
+    string identity;
+    string website;
+    string securityContact;
+    string details;
+
+    Description(string name, string identity, string website, string securityContact,
+                string details)
+        : name(move(name))
+        , identity(move(identity))
+        , website(move(website))
+        , securityContact(move(securityContact))
+        , details(move(details)) {}
+};
+
+#define PRECISION 18
+#define MAX_DECIMAL 1000000000000000000
+
+class Decimal {
```

Comment:
why we need this? just use smallest unit number and type uint256_t

Reply:
or boost/multiprecision/cpp_dec_float.  Implementing own number class seems like reinventing the wheel.

Reply:
The conversion is still needed as the backend expects it. Made the decimal take both value and precision as input.